server/src/main/java/org/elasticsearch/client/node/NodeClient.java

```diff
@@ -91,7 +97,22 @@ void doExecute(ActionType<Response> action, Request request, ActionListener<Resp
                 Response extends ActionResponse
             > Task executeLocally(ActionType<Response> action, Request request, ActionListener<Response> listener) {
         return taskManager.registerAndExecute("transport", transportAction(action), request,
-            (t, r) -> listener.onResponse(r), (t, e) -> listener.onFailure(e));
+                (t, r) -> {
```

Comment:
At least in tests, this ensures no double invocation of the listener with the new `try-catch`
I suppose technically speaking it would be nice to do the same for the override below that uses a `TaskListener` but I chose not to for now since as you can see this already required fixing up a number of spots that relied on the listener's `onResponse` throwing and handling exceptions upstream.